orchagent/muxorch.cpp

```diff
@@ -709,20 +715,44 @@ MuxAclHandler::MuxAclHandler(sai_object_id_t port)
     else
     {
         SWSS_LOG_NOTICE("Binding port %" PRIx64 "", port);
-        // Otherwise just bind ACL table with the port
-        found->second.bind(port);
+
+        AclRule* rule = gAclOrch->getAclRule(table_name, rule_name);
+        if (rule == nullptr)
+        {
+            shared_ptr<AclRuleMux> newRule =
+                    make_shared<AclRuleMux>(gAclOrch, rule_name, table_name, table_type);
+            createMuxAclRule(newRule, table_name);
+        }
+        else
+        {
+            gAclOrch->updateAclRule(table_name, rule_name, MATCH_IN_PORTS, &port, RULE_OPER_ADD);
+        }
     }
 }
 
 MuxAclHandler::~MuxAclHandler(void)
 {
     SWSS_LOG_ENTER();
     string table_name = MUX_ACL_TABLE_NAME;
+    string rule_name = MUX_ACL_RULE_NAME;
 
     SWSS_LOG_NOTICE("Un-Binding port %" PRIx64 "", port_);
 
-    auto found = acl_table_.find(table_name);
-    found->second.unbind(port_);
+    AclRule* rule = gAclOrch->getAclRule(table_name, rule_name);
+    if (rule == nullptr)
+    {
+        SWSS_LOG_THROW("ACL Rule does not exist for port %s, rule %s", alias_.c_str(), rule_name.c_str());
+    }
```

Comment:
Are we guaranteed there will always be a rule here? If there's no PFC storm and no switchover then won't we throw an unnecessary exception?

Reply:
Yes it is guaranteed; In the constructor, we always create an ACL rule and in this destructor, it would be an exception if we don't find the rule.  Same is done for pfwdacl as well, @vmittal-msft,

Reply:
AH this is the handler not the orch, I understand now. Thanks!